### ISSUE TITLE
openPMD: ADIOS2 Engine Options

### ISSUE THREAD

**Issue by ax3l** [ECP-WarpX/WarpX#2866]
Expose ADIOS2 engine options the same way as we expose `adios2_operator`s (#1983).
```
<diag_name>.adios2_operator.type = blosc
<diag_name>.adios2_operator.parameters.compressor = zstd
<diag_name>.adios2_operator.parameters.clevel = 1
<diag_name>.adios2_operator.parameters.doshuffle = BLOSC_BITSHUFFLE
<diag_name>.adios2_operator.parameters.threshold = 2048
<diag_name>.adios2_operator.parameters.nthreads = 6  # per MPI rank (and thus per GPU)
```
This will allow users to control NVM/BurstBuffers and other options.

- NVM/BurstBuffers are engine paramaters in ADIOS2: https://adios2.readthedocs.io/en/latest/engines/engines.html
- This is how we set them as [options](https://github.com/ECP-WarpX/WarpX/blob/716098b54a66b89a107f545f944e48cf7d141ee9/Source/Diagnostics/FlushFormats/FlushFormatOpenPMD.cpp#L63-L84) in openPMD-api:
  - https://openpmd-api.readthedocs.io/en/0.14.4/backends/adios2.html
  - https://openpmd-api.readthedocs.io/en/0.14.4/details/backendconfig.html#backendconfig
  - https://openpmd-api.readthedocs.io/en/0.14.4/details/backendconfig.html#configuration-structure-per-backend
  - https://github.com/openPMD/openPMD-api/blob/1a6bc5308ae21c6adb866882f2fc71de8010bb26/test/ParallelIOTest.cpp#L308-L332

**Comment by kshitij-v-mehta** [comment]
I would like to work on this.

**Comment by ax3l** [comment]
Implemented with #2872. Thanks a ton for your first contribution to WarpX, @kshitij-v-mehta!
Very much appreciated! :rocket: :sparkles:
